File: kinotic-js/workspace/packages/idl/src/api/StreamC3Type.ts
import {C3Type} from '@/api/C3Type'

/**
 * Represents a stream of values produced asynchronously, emitting zero or more results over time.
 * Created by Navíd Mitchell 🤪 on 7/22/26.
 */
export class StreamC3Type extends C3Type {
    /**
     * The type of each value the stream emits
     */
    public valueType: C3Type

    constructor(valueType: C3Type) {
        super('stream')
        this.valueType = valueType
    }
}

File: kinotic-idl/src/main/java/org/kinotic/idl/api/schema/StreamC3Type.java
package org.kinotic.idl.api.schema;

import lombok.*;
import lombok.experimental.Accessors;

/**
 * Represents a stream of values produced asynchronously, emitting zero or more {@link #valueType} results over time.
 * Created by Navíd Mitchell 🤪 on 7/22/26.
 */
@Getter
@Setter
@Accessors(chain = true)
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
@ToString(callSuper = true)
public class StreamC3Type extends C3Type {

    /**
     * The {@link C3Type} of each value the stream emits
     */
    private C3Type valueType = null;

}
